pywhy_stats/discrepancy/kcd.py

```diff
@@ -0,0 +1,261 @@
+"""Kernel (conditional) discrepancy test.
+
+Also known as a conditional k-sample test, where the null hypothesis is that the
+conditional distributions are equal across different population groups.
+
+Returns
+-------
+PValueResult
+    The result of the test, which includes the test statistic and pvalue.
+"""
+
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+from numpy.typing import ArrayLike
+from sklearn.base import BaseEstimator
+from sklearn.preprocessing import LabelBinarizer
+
+from pywhy_stats.kernel_utils import (
+    _default_regularization,
+    _preprocess_kernel_data,
+    compute_kernel,
+)
+
+from ..pvalue_result import PValueResult
+from .base import _compute_propensity_scores, _preprocess_propensity_data, compute_null
+
+
+# XXX: determine if we can do this with Y being optional.
+def condind(
```

Comment:
Here and in the bregman test, an "unconditional" version doesn't make sense right? That would basically mean we just check if two distributions are the same.